chalice/deploy/deployer.py

```diff
@@ -296,22 +297,22 @@ def __init__(self, aws_client):
         # type: (TypedAWSClient) -> None
         self._aws_client = aws_client
 
-    def deploy(self, config, existing_resources):
-        # type: (Config, Optional[DeployedResources]) -> Tuple[str, str, str]
+    def deploy(self, config, existing_resources, lambda_arn):
+        # type: (Config, OPT_RESOURCES, str) -> Tuple[str, str, str]
         if existing_resources is not None and \
                 self._aws_client.rest_api_exists(
                     existing_resources.rest_api_id):
             print "API Gateway rest API already found."
             rest_api_id = existing_resources.rest_api_id
-            return self._create_resources_for_api(config, rest_api_id)
+            return self._create_resources_for_api(config, rest_api_id,
+                                                  lambda_arn)
         else:
             print "Initiating first time deployment..."
-            return self._first_time_deploy(config)
+            return self._first_time_deploy(config, lambda_arn)
```

Comment:
It should always be `lambda_arn` internally in the code.  External references (for example in deployed.json) use `api_handler_arn` because it's referencing the specific lambda function which is the API handler.  Internally in the code, a good chunk of the libraries are generic in the sense that they just require a lambda_arn, they have no context on whether or not it's the api handler arn.

Just to clarify, the PR is to remove `config.lambda_arn` because it's been replaced with all the deployed.json code.